File: 01.backend/src/main/java/com/fallapp/dto/EventoDTO.java
package com.fallapp.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * DTO para transferencia de datos de Evento
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EventoDTO {
    
    @JsonProperty("id_evento")
    private Long idEvento;
    
    @NotNull(message = "El ID de la falla es obligatorio")
    @JsonProperty("id_falla")
    private Long idFalla;
    
    private String nombreFalla;
    
    @NotBlank(message = "El tipo de evento es obligatorio")
    private String tipo;
    
    @NotBlank(message = "El nombre del evento es obligatorio")
    @Size(max = 255, message = "El nombre no puede exceder 255 caracteres")
    private String nombre;
    
    private String descripcion;
    
    @NotNull(message = "La fecha del evento es obligatoria")
    @JsonProperty("fecha_evento")
    private LocalDateTime fechaEvento;
    
    private String ubicacion;
    
    private Integer participantesEstimado;
    
    @JsonProperty("creado_por")
    private Long creadoPor;
    
    @JsonProperty("fecha_creacion")
    private LocalDateTime fechaCreacion;
    
    private String imagenNombre;
}    

File: 03.mobile/app/src/main/java/com/fallapp/features/eventos/data/remote/dto/EventoDto.kt
package com.fallapp.features.eventos.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * DTO de evento según respuesta del backend (campos snake_case).
 * El backend usa @JsonProperty para id_evento, id_falla, fecha_evento.
 */
@Serializable
data class EventoDto(
    @SerialName("id_evento") val idEvento: Long,
    @SerialName("id_falla") val idFalla: Long,
    val nombreFalla: String = "",
    val tipo: String = "",
    val nombre: String = "",
    val descripcion: String? = null,
    @SerialName("fecha_evento") val fechaEvento: String,
    val ubicacion: String? = null,
    val participantesEstimado: Int? = null,
    @SerialName("imagenNombre") val imagen: String? = null
)

